### DISCUSSION TITLE
How to work with `ObjectPath[]`

### DISCUSSION THREAD

**Discussion by Ciavi** [diwic/dbus-rs#453] [Q&A] [answered]
Hello, I am trying to write a _very simple_ app that displays battery percentage on a laptop.
I got to create a connection just fine but then I call:
```
let u_power = dbus_connection.with_proxy("org.freedesktop.UPower", "/org/freedesktop/UPower", Duration::from_secs(5));
let (power_sources,): (Vec<String>,) = u_power.method_call("org.freedesktop.UPower, "EnumerateDevices", ()).unwrap();
```
`power_sources` is empty though. Kind of weird considering that, if I run:
`dbus-send --print-reply --system --dest=org.freedesktop.UPower /org/freedesktop/UPower org.freedesktop.UPower.EnumerateDevices`
it returns:
```
method return time=1698775300.359310 sender=:1.41 -> destination=:1.56 serial=725 reply_serial=2
    array [
        object path "/org/freedesktop/UPower/devices/line_power_AC"
        object path "/org/freedesktop/UPower/devices/battery_BAT0"
        object path "/org/freedesktop/UPower/devices/line_power_ucsi_source_psy_USBC000o001"
        object path "/org/freedesktop/UPower/devices/mouse_dev_D8_CE_2A_67_8E_EC"
    ]
```

Please tell me where I have gone wrong and why! I am fairly new to Rust and I am still learning \:)

**Comment by diwic** [comment, marked as answer]
Use `Vec<dbus::Path>` instead of `Vec<String>`.
